### PR TITLE
bumped up flutter_reaction_button from 2.0.2 to 3.0.0+2.

### LINKED ISSUE #2075: Bump flutter_reaction_button from 2.0.2 to 3.0.0+2
We need to bump [flutter_reaction_button](https://github.com/GeekAbdelouahed/flutter-reaction-button) in the code base from 2.0.2 to 3.0.0+2. 

1. We need to maintain all references to `flutter_reaction_button` in the code base, and you must use the new non-deprecated approaches in this library to maintain the  `flutter_reaction_button` functionality. Deleting flutter_reaction_button functionality is not an option.
2. You must also ensure that all code base tests pass and are valid. Tests without actual executable code does not benefit the community of users, we have experienced this unfortunate fact in the past.
3. The overall code coverage level must be maintained.

<details>
<summary>Release notes for v3.0.0+2</summary>
<p><em>Sourced from <a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/releases">flutter_reaction_button's releases</a>.</em></p>
<blockquote>
<h2>v3.0.0</h2>
<h1>BREAKING CHANGES:</h1>
<ul>
<li><code>FlutterReactionButton</code> and <code>FlutterReactionButtonToggle</code> replaced by <code>ReactionButton</code></li>
<li><code>itemSize</code> is required in <code>ReactionButton</code></li>
<li><code>onReactionChanged</code> callback updated</li>
<li><code>VerticalPosition</code> and <code>HorizontalPosition</code> removed</li>
</ul>
<h1>What's New:</h1>
<ul>
<li>Use overlay instead of navigator</li>
<li>Adds <code>child</code> to <code>ReactionButton</code></li>
<li>Animated <code>ReactionsBox</code></li>
</ul>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/blob/master/CHANGELOG.md">flutter_reaction_button's changelog</a>.</em></p>
<blockquote>
<h2>3.0.0+2</h2>
<ul>
<li>Minor updates</li>
</ul>
<h2>3.0.0+1</h2>
<ul>
<li>Update README</li>
</ul>
<h2>3.0.0</h2>
<ul>
<li><strong>BREAKING CHANGE</strong>:
<ul>
<li><code>FlutterReactionButton</code> and <code>FlutterReactionButtonToggle</code> removed</li>
<li><code>itemSize</code> is required in <code>ReactionButton</code></li>
<li><code>onReactionChanged</code> callback updated</li>
<li><code>VerticalPosition</code> and <code>HorizontalPosition</code> removed</li>
</ul>
</li>
<li>Adds <code>child</code> to <code>ReactionButton</code></li>
<li>Animated <code>ReactionsBox</code></li>
</ul>
</blockquote>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/commit/8132578dd73d4edc37f350014b53e46e9108983b"><code>8132578</code></a> feat: update CHNAGELOG.md</li>
<li><a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/commit/72f2c0bd694793baf8c6eed6f676980d70bd72ac"><code>72f2c0b</code></a> feat: update pubspec.yaml</li>
<li><a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/commit/a8ebd81b313a92775acc246fb7048421ea4e2a6e"><code>a8ebd81</code></a> feat: remove unused resources</li>
<li><a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/commit/a40b15f4529764e298e7a5e13b2aa0fca0df7540"><code>a40b15f</code></a> Merge branch 'feature/overlay' of <a href="https://github.com/GeekAbdelouahed/flutter-">https://github.com/GeekAbdelouahed/flutter-</a>...</li>
<li><a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/commit/439ae8dcbc7a9c19c6e80ea730ab6840ec237423"><code>439ae8d</code></a> feat: update workflows</li>
<li><a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/commit/9e9b501bf0ecfd513e92ce338c6e70426253a685"><code>9e9b501</code></a> Update CHANGELOG.md</li>
<li><a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/commit/37b73f6184222268062a6d107caf56a0fc7924c3"><code>37b73f6</code></a> feat: update pubspec.yaml</li>
<li><a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/commit/1e50dc689e9630afaa429c522f52a4c7c8ad8815"><code>1e50dc6</code></a> feat: update CHANGELOG.md</li>
<li><a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/commit/cb5027636e51e3060f302eaf30f42be169c6eb83"><code>cb50276</code></a> feat: remove mouse region</li>
<li><a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/commit/cfd0234519a34b865c7cafd7c98c3b05b7ab97ed"><code>cfd0234</code></a> feat: remove unused resources</li>
<li>Additional commits viewable in <a href="https://github.com/GeekAbdelouahed/flutter-reaction-button/compare/v2.0.2...v3.0.0">compare view</a></li>
</ul>
</details>
<br />

### PR DESCRIPTION
**Pull request by Azad99-9**
**What kind of change does this PR introduce?**

bug fix: bumped up flutter_reaction_button from 2.0.2 to 3.0.0+2.

**Issue Number:**

Fixes #2075 

**Did you add tests for your changes?**

No

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

Actually this #2075 issue required updation of all references of flutter_reaction_button package in the entire code base

**Does this PR introduce a breaking change?**

No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
Please fix the failing test

**Comment by Azad99-9** [comment]
> Please fix the failing test

files pubspec.yaml and pubspec.lock are listed in the unauthorized files.
In order to update the flutter_reaction_button package, editing of the files pubspec.yaml and pubspec.lock is required.

please have a look and help me with what can be done to pass the respective test.

**Comment by palisadoes** [comment]
> > Please fix the failing test
> 
> files pubspec.yaml and pubspec.lock are listed in the unauthorized files. In order to update the flutter_reaction_button package, editing of the files pubspec.yaml and pubspec.lock is required.
> 
> please have a look and help me with what can be done to pass the respective test.

Accessing the authorized files is understood. You need to fix this error:

- https://github.com/PalisadoesFoundation/talawa/actions/runs/6690122715/job/18194444712?pr=2079

**Comment by Azad99-9** [comment]
> > > Please fix the failing test
> > 
> > 
> > files pubspec.yaml and pubspec.lock are listed in the unauthorized files. In order to update the flutter_reaction_button package, editing of the files pubspec.yaml and pubspec.lock is required.
> > please have a look and help me with what can be done to pass the respective test.
> 
> Accessing the authorized files is understood. You need to fix this error:
> 
> * https://github.com/PalisadoesFoundation/talawa/actions/runs/6690122715/job/18194444712?pr=2079

Yeah, understood.

**Comment by Ayush0Chaudhary** [comment]
@Azad99-9 can you post video-rec of the reaction button?

File: pubspec.lock [hunk 1]
```diff
@@ -729,10 +729,10 @@ packages:
     dependency: transitive
     description:
       name: google_maps_flutter_web
-      sha256: f893d1542c6562bc8299ef768fbbe92ade83c220ab3209b9477ec9f81ad585e4
+      sha256: "05067c5aa762ebee44b7ef4902a311ed8cf891ef655e2798bae063aa3050c8d9"
       url: "https://pub.dev"
     source: hosted
-    version: "0.5.4+2"
+    version: "0.5.4+1"
```
Comment (Ayush0Chaudhary) [inline pubspec.lock:735]: Why did you downgrade this?

File: pubspec.lock [hunk 2]
```diff
@@ -969,10 +969,10 @@ packages:
     dependency: "direct main"
     description:
       name: intl
-      sha256: "3bc132a9dbce73a7e4a21a17d06e1878839ffbf975568bc875c60537824b0c4d"
+      sha256: a3715e3bc90294e971cb7dc063fbf3cd9ee0ebf8604ffeafabd9e6f16abbdbe6
       url: "https://pub.dev"
     source: hosted
-    version: "0.18.1"
+    version: "0.18.0"
```
Comment (Ayush0Chaudhary) [inline pubspec.lock:975]: Why did you downgrade this?

File: pubspec.lock [hunk 3]
```diff
@@ -1486,10 +1486,10 @@ packages:
...
       name: source_span
-      sha256: "53e943d4206a5e30df338fd4c6e7a077e02254531b138a15aec3bd143c1a8b3c"
+      sha256: dd904f795d4b4f3b870833847c461801f6750a9fa8e61ea5ac53f9422b31f250
       url: "https://pub.dev"
     source: hosted
-    version: "1.10.0"
+    version: "1.9.1"
   sqflite:
     dependency: transitive
     description:
```
Comment (Ayush0Chaudhary) [inline pubspec.lock:1495]: Try re-running the `flutter pub get` then again push the changes

**Review by Ayush0Chaudhary — CHANGES_REQUESTED** [review]
Don't downgrade the other packages.

File: pubspec.lock (hunk 1 above)
Comment (Azad99-9) [reply pubspec.lock:735]: I did not intentionally downgrade any packages. The downgrade occurred automatically right after I updated the `reaction_button` package in the `pubspec.yaml` file.

Comment (Azad99-9) [reply pubspec.lock:735]: fixed the package downgrading issue.

**Comment by Azad99-9** [comment]
> @Azad99-9 can you post video-rec of the reaction button?


https://github.com/PalisadoesFoundation/talawa/assets/120930148/49770e6e-879e-4ec6-bd27-5bafce39435c

video of reaction button.

**Comment by palisadoes** [comment]
Please fix the failing test

**Comment by Ayush0Chaudhary** [comment]
Test are failing, fix them please!

**Comment by Azad99-9** [comment]
@palisadoes @Ayush0Chaudhary 
I have worked on fixing the failing tests and i found out that, the test file that is failing has following lines of code:
`await Hive.openBox<User>('currentUser');
  await Hive.openBox<OrgInfo>('currentOrg');
  await Hive.openBox('url');`
these lines does not impact the actual test logic in that file : talawa/test/widget_tests/after_auth_screens/app_settings/app_setting_page_test.dart.

So i tried removing those lines and ran the tests and all the tests were passing.
Shall i commit the changes that were made to the test file.

**Comment by Ayush0Chaudhary** [comment]
> @palisadoes @Ayush0Chaudhary 
> I have worked on fixing the failing tests and i found out that, the test file that is failing has following lines of code:
> `await Hive.openBox<User>('currentUser');`
> `await Hive.openBox<OrgInfo>('currentOrg');`
> `await Hive.openBox('url');`
> these lines does not impact the actual test logic in that file : talawa/test/widget_tests/after_auth_screens/app_settings/app_setting_page_test.dart.
> 
> So i tried removing those lines and ran the tests and all the tests were passing.
> Shall i commit the changes that were made to the test file.

Yes

**Comment by noman2002** [comment]
@Azad99-9 Please fix the conflicting files.

**Comment by Azad99-9** [comment]
@noman2002 Please advise on resolving the 'unauthorized files' error.

**Comment by palisadoes** [comment]
> @noman2002 Please advise on resolving the 'unauthorized files' error.

The test is failing because you have edited one or more files that we more carefully track to reduce the risk of breaking our GitHub action workflows and other related automation. This is to be expected in certain types of PRs especially those that alter the packages required by the repo’s app.

**Comment by Azad99-9** [comment]
> > Please fix the failing test
> 
> files pubspec.yaml and pubspec.lock are listed in the unauthorized files. In order to update the flutter_reaction_button package, editing of the files pubspec.yaml and pubspec.lock is required.
> 
> please have a look and help me with what can be done to pass the respective test.

@palisadoes please have a look at this.

**Review by noman2002 — APPROVED** [review]
Looks good to me.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**